pkg/fetcher/fetcher.go

```diff
@@ -100,6 +100,21 @@ type URLFetcher struct {
 	options Options
 }
 
+// ErrorResponseBody struct
```

Comment:
Does `PullErrorRespBody` mean error response for pull request? It is not necessarily for pull request only (convention: https://docs.docker.com/registry/spec/api/#errors) tho it could be that it's only used when pulling from registry tho